pyfolio/risk.py

```diff
@@ -0,0 +1,486 @@
+#
+# Copyright 2017 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+import matplotlib.pyplot as plt
+from collections import OrderedDict
+
+SECTORS = OrderedDict([
+    (101, '101 Basic Materials'),
+    (102, '102 Consumer Cyclical'),
+    (103, '103 Financial Services'),
+    (104, '104 Real Estate'),
+    (205, '205 Consumer Defensive'),
+    (206, '206 Healthcare'),
+    (207, '207 Utilities'),
+    (308, '308 Communication Services'),
+    (309, '309 Energy'),
+    (310, '310 Industrials'),
+    (311, '311 Technology')
+])
+
+CAP_CUTOFFS = [50000000, 300000000, 2000000000, 10000000000, 200000000000]
+CAP_NAMES = ['Micro', 'Small', 'Mid', 'Large', 'Mega']
+
+
+def compute_style_factor_exposures(positions, risk_factor):
+    '''
+    Returns style factor exposure of an algorithm's positions
+
+    Parameters
+    ----------
+    positions : pd.DataFrame
+        Daily equity positions of algorithm, in dollars.
+        - See full explanation in create_risk_tear_sheet
+
+    risk_factor : pd.DataFrame
+        Daily risk factor per asset.
+        - DataFrame with dates as index and equities as columns
+        - Example:
+                     Equity(24   Equity(62
+                       [AAPL])      [ABT])
+        2017-04-03	  -0.51284     1.39173
+        2017-04-04	  -0.73381     0.98149
+        2017-04-05	  -0.90132	   1.13981
+    '''
+
+    positions_wo_cash = positions.drop('cash', axis=1)
+    gross_exposure = positions_wo_cash.abs().sum(axis=1)
+
+    sfe = positions_wo_cash.multiply(risk_factor, level=1) \
```

Comment:
If these are mult-index, the doc string should reflect it. Moreover, multi-indexes should be named to give `level='factor'`.